tests/utils/common/common.py

```diff
@@ -32,6 +28,79 @@
 import traceback
 
 
+class Result:
+    def __init__(self, stdout, stderr, exited):
+        self.stdout = stdout
+        self.stderr = stderr
+        self.exited = exited
+        self.return_code = exited
+
+
+class Connection:
+    def __init__(self, host, user, port, connect_timeout, connect_kwargs={}):
+        self.host = host
+        self.user = user
+        self.port = port
+        self.connect_timeout = connect_timeout
+        self.connect_kwargs = connect_kwargs
+
+        self.key_filename = None
+
+        for k in connect_kwargs.keys():
+            if k == "key_filename":
+                self.key_filename = connect_kwargs[k]
+            else:
+                raise NotImplementedError(f"Argument {k} is not implemented")
+
+    def get_connect_args(self):
+        if self.key_filename is not None:
+            key_arg = ["-i", self.key_filename]
+        else:
+            key_arg = []
+
+        args = (
```

Comment:
Q: Why the surrounding brackets?

Reply:
I don't know, Black did this. It's probably because the `+` signs don't have any parentheses, so to use line breaks, there has to be a pair around the outermost expression.

Reply:
makes sense